packages/forms/resources/views/components/builder/group-block-picker.blade.php

```diff
@@ -0,0 +1,205 @@
+@props([
+    'action',
+    'afterItem' => null,
+    'groups' => [],
+    'blocks',
+    'columns' => null,
+    'statePath',
+    'trigger',
+    'width' => null,
+])
+
+@php
+    $blocksByGroupName = collect($blocks)
+        ->groupBy(fn (Filament\Forms\Components\Builder\Block $block) => $block->getGroup())
+        ->all();
+
+    $groups = collect($groups)
+        ->filter(fn (Filament\Forms\Components\Builder\BlockGroup $group) => isset($blocksByGroupName[$group->getName()]))
+        ->all();
+
+    $hasUngroupedBlocks = isset($blocksByGroupName['']);
+@endphp
+
+<x-filament::dropdown
+    :width="$width"
+    {{ $attributes->class(['fi-fo-builder-block-picker']) }}
+>
+    <x-slot name="trigger">
+        {{ $trigger }}
+    </x-slot>
+
+    <x-filament::dropdown.list
+        x-data="{
+        group: null,
+    }"
+    >
+        <div x-show="=== null group">
+            @foreach ($groups as $group)
+                @php
+                    $xOnClickAction = 'group = ' . Js::from($group->getName());
+                @endphp
+
+                <x-filament::dropdown.list.item
+                    :icon="$group->getIcon()"
+                    :icon-size="$group->getIconSize()"
+                    :x-on:click="$xOnClickAction"
+                >
+                    <div class="flex items-center justify-between">
+                        <div class="space-y-1">
+                            <p class="font-medium">
+                                {{ $group->getLabel() }}
+                            </p>
+
+                            @if ($description = $group->getDescription())
```

Comment:
Potentially we could support description and suffix icon in the dropdown list item component instead?

Reply:
Yeah, like I said, it's functional but needs tidy up and design love.